python/lsst/pipe/base/graph/_versionDeserializers.py

```diff
@@ -0,0 +1,524 @@
+        end of the metadata.
+        """
+        raise NotImplementedError("Base class does not implement this method")
+
+    def readHeaderInfo(self, rawHeader: bytes) -> SimpleNamespace:
+        """Parse the supplied raw bytes into the header information and
+        byte ranges of specific TaskDefs and QuantumNodes
+
+        rawheader : bytes
+            The bytes that are to be parsed into the header information. These
+            are the bytes after the preamble and structsize number of bytes
+            and before the headerSize bytes
+        """
+        raise NotImplementedError("Base class does not implement this method")
+
+    def constructGraph(self, nodes: set[uuid.UUID], _readBytes: Callable,
+                       universe: DimensionUniverse) -> QuantumGraph:
+        raise NotImplementedError("Base class does not implement this method")
+
+    def description(self) -> str:
+        """Return the description of the serialized data format
+        """
+        raise NotImplementedError("Base class does not implement this method")
+
+
+Version1Description = """
+The save file starts with the first few bytes corresponding to the magic bytes
+in the QuantumGraph: `qgraph4\xf6\xe8\xa9`.
+
+The next few bytes are 2 big endian unsigned long longs.
+
+The first unsigned long long corresponds to the number of bytes of a python
+mapping of TaskDef labels to the byte ranges in the save file where the
+definition can be loaded.
+
+The second unsigned long long corrresponds to the number of bytes of a python
+mapping of QuantumGraph Node number to the byte ranges in the save file where
+the node can be loaded. The byte range is indexed starting after
+the `header` bytes of the magic bytes, size bytes, and bytes of the two
+mappings.
+
+Each of the above mappings are pickled and then lzma compressed, so to
+deserialize the bytes, first lzma decompression must be performed and the
+results passed to python pickle loader.
+
+As stated above, each map contains byte ranges of the corresponding
+datastructure. Theses bytes are also lzma compressed pickles, and should
+be deserialized in a similar manor. The byte range is indexed starting after
+the `header` bytes of the magic bytes, size bytes, and bytes of the two
+mappings.
+
+In addition to the the TaskDef byte locations, the TypeDef map also contains
+an additional key '__GraphBuildID'. The value associated with this is the
+unique id assigned to the graph at its creation time.
+"""
+
+
+@dataclass
+class DeserializerV1(DeserializerBase):
+    FMT_STRING: ClassVar[str] = '>QQ'
+
+    def __post_init__(self):
+        self.taskDefMapSize, self.nodeMapSize = struct.unpack(self.FMT_STRING, self.sizeBytes)
+
+    @property
+    def headerSize(self) -> int:
+        return self.preambleSize + self.structSize + self.taskDefMapSize + self.nodeMapSize
+
+    def readHeaderInfo(self, rawHeader: bytes) -> SimpleNamespace:
+        returnValue = SimpleNamespace()
```

Comment:
I considered it, and dont have strong feelings. I think dataclasses give a bit more schema definition, but I was not really ready to lock into something in case future save formats would benefit from some flexibility. Though I admit I cant really nail down a good reason not to, was just a general feeling

Reply:
:+1:, something we can revisit later, then.